desktop/lib/app-quit/index.js

```diff
@@ -10,12 +10,12 @@ function AppQuit() {
 }
 
 AppQuit.prototype.shouldQuitToBackground = function() {
-	if ( this.canQuit === false ) {
-		return true;
+	if ( this.canQuit ) {
+		this.canQuit = false;
+		return false;
 	}
 
-	this.canQuit = false;
-	return false;
+	return true;
 };
```

Comment:
Sorry, I miss what we are changing here 🤔

Reply:
Thanks for the review/testing, @loremattei!

> I miss what we are changing here 

You're right - in the case of `shouldQuitToBackground`, I think the code change winds up with the same result. However, I found it challenging to reason about the logic as-written.

In addition, on Windows, the application was _not_ closing the main window, leading to a duplicate installation of the app. There were two methods of concern around how the application handles backgrounding and the `close` event (which is emitted by `app.quitAndInstall`).

```
AppQuit.prototype.shouldQuitToBackground = function() {
	if ( this.canQuit === false ) {
		return true;
	}

	this.canQuit = false;
	return false;
};
```

```
WindowsPlatform.prototype.onClosed = function( ev ) {
	if ( appQuit.shouldQuitToBackground() ) {
		log.info( 'Window close puts app into background & creates tray' );

		ev.preventDefault();

		this.window.hide();
		this.showBackgroundBubble();
	}
};
```

I found the above a little difficult to reason about, and refactored things slightly to make sure that each code path is explicitly handled as well as make things a little easier to grok.

Reply:
👍